File: DTOs/AppointmentRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace MediCare.DTOs
{
	public class AppointmentRequestDTO
	{
		[Required(ErrorMessage = "Appointment date is required")]
		[DataType(DataType.DateTime, ErrorMessage = "Invalid date-time format")]
		public DateTime Time { get; set; }

		public int? DoctorId { get; set; }

		public int? ServiceId { get; set; }
	}
}

File: MediCareWebSpa/src/app/DTOs/appointment-request.dto.ts
export class AppointmentRequestDTO {
  time: string;
  doctorId?: number;
  serviceId?: number;

  constructor(time: string, doctorId?: number, serviceId?: number) {
    this.time = time;
    this.doctorId = doctorId;
    this.serviceId = serviceId;
  }
}
